File: src/models/ContactRequest.ts
import mongoose, { Schema, Model } from 'mongoose';
import { IContactRequest } from '../types';

const contactRequestSchema = new Schema<IContactRequest>(
    {
        name: {
            type: String,
            required: true,
        },
        email: {
            type: String,
            required: true,
        },
        message: {
            type: String,
            required: true,
        },
        status: {
            type: String,
            enum: ['pending', 'resolved'],
            default: 'pending',
        },
    },
    {
        timestamps: true,
    }
);

// Create indexes
contactRequestSchema.index({ status: 1, createdAt: -1 });
contactRequestSchema.index({ email: 1 });

export const ContactRequest: Model<IContactRequest> = mongoose.model<IContactRequest>(
    'ContactRequest',
    contactRequestSchema
);

File: src/main/java/com/medicare/backend/models/ContactRequest.java
package com.medicare.backend.models;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "contact_requests")
public class ContactRequest {
    @Id
    private String id;

    private String name;

    @Indexed
    private String email;

    private String message;

    @Indexed
    private String status = "pending"; // pending, resolved

    @CreatedDate
    private LocalDateTime createdAt;

    @LastModifiedDate
    private LocalDateTime updatedAt;
}
